src/storage.cpp

```diff
@@ -1692,6 +1693,9 @@ namespace libtorrent
 	{
 		TORRENT_ASSERT(m_files.piece_length() > 0);
 
+		if (checked_fastresume) return check_no_fastresume(ec);
```

Comment:
continuing down here, checking the resume data, isn't supposed to invalidate it. do you have a reason to believe it does? as far as I can tell, you should be able to call this function multiple times and have the same outcome. If you restore the logic here and run your test, do you get any errors?